### ISSUE TITLE
Clarify wording in deprecation message for reactor.spawnProcess

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8799]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8799|
|Type|enhancement|
|Created|2016-08-29 09:39:47Z|

This message:
```
 Argument strings and environment keys/values passed to reactor.spawnProcess should be str, not unicode
```

should mention **bytes** not **str** to avoid confusion on Python 3.

<details><summary>Searchable metadata</summary>

```
trac-id__8799 8799
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1472463587564626 1472463587564626
changetime__1472479315298882 1472479315298882
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="derwolfe's avatar" src="https://avatars.githubusercontent.com/u/641357?s=50" width="50" height="50">](https://github.com/derwolfe)<a name="note_1"></a>|@derwolfe set owner to @rodrigc|
|-|-|

This looks good to land. Please feel free to merge once CI is passing - thanks :)

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_2"></a>|@rodrigc set status to `closed`|
|-|-|

In changeset 60f1fc2b21c601338fab2e3442dd0f976e90d2cf
```
#!CommitTicketReference repository="" revision="60f1fc2b21c601338fab2e3442dd0f976e90d2cf"
Merge pull request #531 from twisted/8799-rodrigc-reactorSpawnProcess

Author: rodrigc
Reviewer: derwolfe, markrwilliams
Fixes: #8799

Change 'str' to 'bytes' in deprecation message
```
